code/game/objects/empulse.dm

```diff
@@ -15,18 +16,42 @@
 	if(heavy_range > light_range)
 		light_range = heavy_range
 
-	for(var/A in spiral_range(light_range, epicenter))
-		var/atom/T = A
-		var/distance = get_dist(epicenter, T)
+	var/turf/new_location_above = epicenter
+	var/turf/new_location_below = epicenter
+	var/base_level_pulsed = FALSE
+	for(var/levels in 1 to EMPULSE_MAX_LEVELS_AFFECT)
+		if(!base_level_pulsed)
+			empulse_call(epicenter, heavy_range, light_range)
+			base_level_pulsed = TRUE
+
+		heavy_range *= 0.25
+		light_range *= 0.25
+		var/turf/above_level = SSmapping.get_turf_above(new_location_above)
+		var/turf/below_level = SSmapping.get_turf_below(new_location_below)
+
+		if(above_level && isturf(above_level))
+			empulse_call(above_level, heavy_range, light_range)
+			new_location_above = above_level
+		if(below_level && isturf(below_level))
+			empulse_call(below_level, heavy_range, light_range)
+			new_location_below = below_level
```

Comment:
maybe considering decreasing the intensity as it travels through zlevels by some coefficient

Reply:
`heavy_range *= 0.25`
`light_range *= 0.25`
already done that